### DISCUSSION TITLE
How to deal with it Scoop refreshing PATH and erasing variables references?

### DISCUSSION THREAD

**Discussion by nonatorw** [ScoopInstaller/Scoop#5059] [Ideas]
I recently started to use `scoop` at my work computer, and I have  environment variables that I need to include in the `PATH` to use some tools that `scoop` can't manage. However, every time that I run `scoop update --all` or `scoop cleanup --all`, all my variables included in the `PATH` are changed for their absolute paths. It is troublesome because some of that variables need to be changed at runtime.

So, I don't know the impact of this suggestion to the whole project, but... why `scoop` doesn't use environment variables (that can be included in the `PATH`) and update them, instead of put the applications' paths directly in the `PATH`, refreshing it and killing every variable's reference?

Individual environment variables for each app, depending on the quantity of apps to include in the environment, could be troublesome to manage. However, working in that way could avoid problems of compatibility with other apps that are not managed by `scoop`.

**A complementar suggestion**: to declare a `SCOOP_APPS_PATH` variable, and include all `scoop` managed's apps in it, and then do reference to it in the `PATH` variable.

**Comment by rashil2000** [comment]
Hi, could you post your feedback in the issue tracker - here https://github.com/ScoopInstaller/Scoop/issues/1813?

**Reply by nonatorw** [reply]
Done.

**Comment by r15ch13** [comment]
Okay, it seems we are destroying users `PATH` variables [since the dawn of time](https://github.com/ScoopInstaller/Scoop/blob/aa3826c23687e381f0eac05808a4189a90b82c56/lib/init.ps1): https://github.com/ScoopInstaller/Scoop/blob/aa3826c23687e381f0eac05808a4189a90b82c56/lib/init.ps1#L6-L9

Brought to attention via https://github.com/ScoopInstaller/Scoop/pull/5395
Related: https://github.com/ScoopInstaller/Scoop/issues/1813
Samples for handling this issue:
- [choco Update-SessionEnvironment.ps1](https://github.com/chocolatey/choco/blob/stable/src/chocolatey.resources/helpers/functions/Update-SessionEnvironment.ps1)
- [choco Get-EnvironmentVariable.ps1](https://github.com/chocolatey/choco/blob/stable/src/chocolatey.resources/helpers/functions/Get-EnvironmentVariable.ps1)
- [choco Get-EnvironmentVariableNames.ps1](https://github.com/chocolatey/choco/blob/stable/src/chocolatey.resources/helpers/functions/Get-EnvironmentVariableNames.ps1)
- [choco Set-EnvironmentVariable.ps1](https://github.com/chocolatey/choco/blob/stable/src/chocolatey.resources/helpers/functions/Set-EnvironmentVariable.ps1)


This needs to be implemented in the installer and Scoop itself:
- Should handle `%%` Variables with and without resolving them
- Update User env after changes (e.g. installing Scoop itself and programs)
- Adding a `SCOOP_PATH` variable to free up space in `PATH`
